### DISCUSSION TITLE
Collection Literals - Ignore Null Items/Ranges

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7351] [Language Ideas] [answered]
I would like to be able to create a collection literal that ignores null items and ranges.
Here is how I think it should look:
```
var items = [ NonNullItem, ?PotentiallyNullItem, ..NonNullCollection, ?..PotentiallyNullCollection];
```

**Comment by 333fred** [comment, marked as answer]
I believe the current plan is something like `[ .. (potentiallyNullCollection ?? [])]`, where it would be efficiently emitted.

**Reply by TonyValenti** [reply]
Is there a way to omit single elements as well?

**Reply by CyrusNajmabadi** [reply]
Can you given an example of what you mean @TonyValenti ?

**Reply by 333fred** [reply]
@CyrusNajmabadi it was in the original post that I missed: `var items = [ NonNullItem, ?PotentiallyNullItem];`

I believe the intent is that you'd wrap that in a single element, spread that, and it would be efficiently emitted, but I'm unsure.

**Reply by RikkiGibson** [reply]
`[..elem != null ? [elem] : []]` is the pattern I would expect for conditional inclusion of elements. I'm certainly interested to see if there's demand for more concise patterns here after we ship.

**Reply by TonyValenti** [reply]
I just wanted to post here and say I'd really like a concise syntax to ignore null items.

Today, I'm adding all the null items into the collection and then using a ```.WhereIsNotNull()``` extension method on the collection.
It isnt as efficient, but the code looks a lot prettier.
